packages/esm-patient-common-lib/src/workspaces/workspaces.ts

```diff
@@ -187,6 +189,7 @@ export function launchPatientWorkspace(name: string, additionalProps?: object) {
   const openWorkspaces = store.getState().openWorkspaces;
   const workspaceIndexInOpenWorkspaces = openWorkspaces.findIndex((w) => w.name === name);
   const isWorkspaceAlreadyOpen = workspaceIndexInOpenWorkspaces >= 0;
+  const isEditingEncounter = newWorkspace?.additionalProps?.hasOwnProperty('formInfo');
```

Comment:
Can we have something better to have a comparision that the workspace is being opened with different state?

Reply:
No, unfortunately, or maybe you have some suggestions?

Reply:
Pulling in @denniskigen for suggestion here.
Thanks!

Reply:
@denniskigen @vasharma05 any updates?

Reply:
const isEditingEncounter = !!newWorkspace?.additionalProps?.formInfo;
(nullish coalescing approach + optional chanining method) try with this. but the time complexity seems to be similar for both.